src/packs.c

```diff
@@ -199,7 +223,64 @@ int download_subscribed_packs(struct list *subs, struct manifest *mom, bool requ
 			}
 		}
 	}
-	progress_report(list_length, list_length); /* Force out 100% */
 	info("\n");
 	return swupd_curl_parallel_download_end(download_handle, NULL);
 }
+
+double packs_query_total_download_size(struct list *subs, struct manifest *mom)
+{
+	double size = 0;
+	double total_size = 0;
+	struct sub *sub = NULL;
+	struct list *list = NULL;
+	struct file *bundle = NULL;
+	char *targetfile = NULL;
+	char *url = NULL;
+	struct stat stat;
+	int count = 0;
+
+	for (list = list_head(subs); list; list = list->next) {
+		sub = list->data;
+
+		if (sub->oldversion == sub->version) {
```

Comment:
This code is duplicated from download_pack. This is hard to maintain. If we change the strategy on when to download packs we would need to change in both functions.

Reply:
this duplication was removed in a latter commit